code/modules/mob/mob_movement.dm

```diff
@@ -175,6 +175,9 @@
 	if(mob && .)
 		mob.throwing = 0
 
+	for(var/obj/O in mob)
+		O.on_mob_move(direct, src)
```

Comment:
ew, ew ew ew; you could get around having to do this with shoes(which you have!!) and proper abuse of step_action()
it'd be gnarly but maybe better